fv3core/stencils/fxadv.py

```diff
@@ -1,1026 +1,861 @@
-                sin_sg2,
-                sin_sg4,
-                vt,
-                dt=dt,
-                origin=(0, grid().je + 1, 0),
-                domain=edge_shape,
-            )
+@gtscript.function
+def ra_x_func(area, xfx_adv):
+    return area + xfx_adv - xfx_adv[1, 0, 0]
 
 
-# -------------------- CORNERS-----------------
+@gtscript.function
+def ra_y_func(area, yfx_adv):
+    return area + yfx_adv - yfx_adv[0, 1, 0]
 
 
-def corner_ut_stencil(
+"""
+@gtstencil()
+def fxadv_stencil(
+    cosa_u: FloatFieldIJ,
+    cosa_v: FloatFieldIJ,
+    rsin_u: FloatFieldIJ,
+    rsin_v: FloatFieldIJ,
+    sin_sg1: FloatFieldIJ,
+    sin_sg2: FloatFieldIJ,
+    sin_sg3: FloatFieldIJ,
+    sin_sg4: FloatFieldIJ,
     uc: FloatField,
     vc: FloatField,
     ut: FloatField,
     vt: FloatField,
-    cosa_u: FloatFieldIJ,
-    cosa_v: FloatFieldIJ,
+    dt: float,
 ):
-    from __externals__ import ux, uy, vi, vj, vx, vy
+    Updates flux operators and courant numbers for fvtp2d
+
+    To kick off D_SW after the C-grid winds have been advanced half a timestep,
+    and and compute finite volume transport on the D-grid (e.g.Putman and Lin 2007),
+    this module prepares terms such as parts of equations 7 and 13 in Putnam and Lin,
+    2007, that get consumed by fvtp2d and ppm methods.
+
+    Args:
+        uc: x-velocity on the C-grid (in)
+        vc: y-velocity on the C-grid (in)
+        crx_adv: Courant number, x direction(inout)
+        cry_adv: Courant number, y direction(inout)
+        xfx_adv: Finite volume flux form operator in x direction (inout)
+        yfx_adv: Finite volume flux form operator in y direction (inout)
+        ut: temporary x-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        vt: temporary y-velocity transformed from C-grid to D-grid equivalent(?) (inout)
+        dt: timestep in seconds
+    Grid variable inputs:
+        cosa_u, cosa_v, rsin_u, rsin_v, sin_sg1,sin_sg2, sin_sg3, sin_sg4
 
     with computation(PARALLEL), interval(...):
-        ut[0, 0, 0] = (
-            (
-                uc[0, 0, 0]
-                - 0.25
-                * cosa_u[0, 0]
-                * (
-                    vt[vi, vy, 0]
-                    + vt[vx, vy, 0]
-                    + vt[vx, vj, 0]
-                    + vc[vi, vj, 0]
-                    - 0.25
-                    * cosa_v[vi, vj]
-                    * (ut[ux, 0, 0] + ut[ux, uy, 0] + ut[0, uy, 0])
-                )
-            )
-            * 1.0
-            / (1.0 - 0.0625 * cosa_u[0, 0] * cosa_v[vi, vj])
-        )
-
+        ut = main_ut(uc, vc, cosa_u, rsin_u, ut)
+        ut = ut_y_edge(uc, sin_sg1, sin_sg3, ut, dt)
+        vt = main_vt(uc, vc, cosa_v, rsin_v, vt)
+        vt = vt_y_edge(vc, cosa_v, ut, vt)
+        vt = vt_x_edge(vc, sin_sg2, sin_sg4, vt, dt)
+        ut = ut_x_edge(uc, cosa_u, vt, ut)
+        ut = ut_corners(uc, vc, cosa_u, cosa_v, ut, vt)
+        vt = vt_corners(uc, vc, cosa_u, cosa_v, ut, vt)
+"""
```

Comment:
Why is this code block-commented?

Reply:
This is what the single stencil version is, all the things that are stencils now can be functions with the numpy backend, so when the gtbackends support the regions craziness we can use this